src/core/shards/core.hpp

```diff
@@ -3469,7 +3469,7 @@ struct Once {
         // let's cheat in this case and stop triggering this call
         self->inlineShardId = InlineShard::NoopShard;
       } else {
-        next = next + dsleep;
+        next = (current - (current - next)) + dsleep;
```

Comment:
This is just the same but more calculations 😄

Reply:
not exactly, the above took no care of `current` at all in fact was broken.
did you check above?

Reply:
`current - next` is the drift which ideally is 0 but in reality never is 0
We subtract it from current to remove how much we drifted and add the sleep.
I think the logic is not flawed here to try to realign to real time

Reply:
I mean it's mathematically equivalent: `(current - (current - next)) == next` so it still evaluates to `next + dsleep`

Reply:
https://www.wolframalpha.com/input?i=c+-+%28c+-+n%29+%2B+d

Reply:
Oh you mean canceling each other. How do you explain that the above is broken though when used 🤣🤔? That's why I had to fix it.